lib/developer_portal/lib/liquid/forms/account.rb

```diff
@@ -8,7 +8,13 @@ def html_class_name
         end
 
         def path
-          admin_account_payment_details_path
+          polymorphic_path([:admin, :account, payment_gateway_type])
```

Comment:
@jlledom I applied the change that makes the `#update` to always get executed within a subclass controller.

I've tested the flows though, and it turns out that Stripe is the only one that reaches that path.

Braintree has a specific form that joins billing address and credit card, and Authorize.net has a completely different form. When the details get updated they are first saved on the payment gateway, and then in our DB through `hosted_success` method.

References: edit form only shows for [stripe and braintree](https://github.com/3scale/porta/blob/f7edfc3284db96a1a0ac1f34bcad2bcef8551d8b/lib/developer_portal/app/views/developer_portal/accounts/payment_gateways/show.html.liquid#L15-L33). And the billing address form is only rendered for [stripe](https://github.com/3scale/porta/blob/f7edfc3284db96a1a0ac1f34bcad2bcef8551d8b/lib/developer_portal/app/views/developer_portal/accounts/payment_gateways/edit.html.liquid#L25-L38).

So, `BraintreeBlueController#update` and `AuthorizeNetController#update` should never be used in standard flow. But as this `account.billing_address` form can be added somewhere else by the customer manually, I think it's still reasonable to support the flow for all gateways, and it should work as before.

Reply:
Thanks for the info 👍